Rollup of 7 pull requests

kennytm opened the pull request:
Successful merges:

 - #49707 (Add "the Rustc book")
 - #50222 (Bump bootstrap compiler to 2018-04-24)
 - #50227 (Fix ICE with erroneous `impl Trait` in a trait impl)
 - #50229 (Add setting to go to item if there is only one result)
 - #50231 (Add more doc aliases)
 - #50246 (Make dump_{alloc,allocs,local}() no-ops when tracing is disabled.)
 - #49894 (Rename InternedString to LocalInternedString and introduce a new thread-safe InternedString)

Failed merges:

bors commented:
:broken_heart: Test failed - [status-appveyor](https://ci.appveyor.com/project/rust-lang/rust/build/1.0.7174)

kennytm commented:
@bors r+

Removed #49891 (already testing)
Removed #50252 (I suspect the compile segfault during `stage1-std` on Windows is caused by this)
Added #49894 (it is the next one in queue)

bors commented:
:pushpin: Commit 44b5359 has been approved by `kennytm`

<!-- @bors r=kennytm 44b53598507b88bd256477cb4717ec6817ffbcec -->

bors commented:
:hourglass: Testing commit 44b53598507b88bd256477cb4717ec6817ffbcec with merge ada45fd49a2dfcf2a163e1f4641ac34f2803eb9b...

bors commented:
:sunny: Test successful - [status-appveyor](https://ci.appveyor.com/project/rust-lang/rust/build/1.0.7176), [status-travis](https://travis-ci.org/rust-lang/rust/builds/371956047?utm_source=github_status&utm_medium=notification)
Approved by: kennytm
Pushing ada45fd49a2dfcf2a163e1f4641ac34f2803eb9b to master...
